wrangler-core/src/test/java/io/cdap/functions/DateAndTimeTest.java

```diff
@@ -0,0 +1,521 @@
+/*
+ *  Copyright © 2020 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package io.cdap.functions;
+
+import io.cdap.wrangler.TestingRig;
+import io.cdap.wrangler.api.Row;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A class to test {@link DateAndTime} functions.
+ */
+public class DateAndTimeTest {
+
+  @Test
+  public void testCurrentDateTime() throws Exception {
+    String[] directives = new String[]{
+      "set-column CurrentDate datetime:CurrentDate()",
+      "set-column CurrentTime datetime:CurrentTime()",
+      "set-column CurrentTimeMS datetime:CurrentTimeMS()",
+      "set-column CurrentTimestamp datetime:CurrentTimestamp()",
+      "set-column CurrentTimestampMS datetime:CurrentTimestampMS()",
+      "set-column CurrentTimestampMS datetime:CurrentTimestampMS()",
+    };
+
+    List<Row> rows = Arrays.asList(new Row());
+    rows = TestingRig.execute(directives, rows);
+    Assert.assertTrue(rows.size() == 1);
+    Assert.assertNotNull(rows.get(0).getValue("CurrentDate"));
+    Assert.assertNotNull(rows.get(0).getValue("CurrentTime"));
+    Assert.assertNotNull(rows.get(0).getValue("CurrentTimeMS"));
+    Assert.assertNotNull(rows.get(0).getValue("CurrentTimestamp"));
+    Assert.assertNotNull(rows.get(0).getValue("CurrentTimestampMS"));
+    Assert.assertNotNull(rows.get(0).getValue("CurrentTimestampMS"));
+  }
+
+  @Test
+  public void testDateFromDaysSince() throws Exception {
+    String[] directives = new String[]{
+      "set-column DateFromDaysSince1 datetime:DateFromDaysSince(18250, datetime:GetDate('1958-08-18'))", // 2008-08-05
+      "set-column DateFromDaysSince2 datetime:DateFromDaysSince(-1, datetime:GetDate('1958-08-18'))", // 1958-08-17
+    };
+
+    List<Row> rows = Arrays.asList(new Row());
+    rows = TestingRig.execute(directives, rows);
+    Assert.assertTrue(rows.size() == 1);
+    Assert.assertEquals(DateAndTime.GetDate("2008-08-05"), rows.get(0).getValue("DateFromDaysSince1"));
```

Comment:
should create the expected object directly instead of calling the library that is being tested.